src/main/java/programminglife/model/drawing/SubGraph.java

```diff
@@ -285,7 +285,7 @@ public boolean checkDynamicLoad(int leftBorder, double rightBorder) {
      * @return int of the new centerNode.
      */
     public int updateCenterNode(double centerCanvasX, int oldCenterNode) {
-        for (int i = 0; i < layers.size(); i++) {
+        for (int i = 1; i < layers.size(); i++) {
             if (layers.get(i).getX() > centerCanvasX) {
                 if (Math.abs(layers.get(i - 1).getX() - centerCanvasX)
                         < Math.abs(layers.get(i).getX() - centerCanvasX)) {
```

Comment:
instead of doing this, you can also use `Layer centerLayer = layers.get(centerCanvasX)`